From f8364104fa7266735fb08e50d988f6e80acadf7a Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 22 Oct 2023 10:31:23 +0100
Subject: [PATCH 1/1] data: Tweak systemd network targets to get offline update
 working reliably

Solus, downstream only.

Even when an offline transation is pre-downloaded pk still wants network :(

We were hitting an issue where pk offline update would try to start
the transaction before networking was up.
---
 data/packagekit-offline-update.service.in | 1 +
 data/packagekit.service.in                | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/data/packagekit-offline-update.service.in b/data/packagekit-offline-update.service.in
index 23096058f..b2c19cd03 100644
--- a/data/packagekit-offline-update.service.in
+++ b/data/packagekit-offline-update.service.in
@@ -7,6 +7,7 @@ After=sysinit.target dbus.socket systemd-journald.socket system-update-pre.targe
 Before=shutdown.target system-update.target
 # See packagekit.service
 ConditionPathExists=!/run/ostree-booted
+Wants=network-online.target
 
 [Service]
 Type=oneshot
diff --git a/data/packagekit.service.in b/data/packagekit.service.in
index 52884589a..0a01d3a0c 100644
--- a/data/packagekit.service.in
+++ b/data/packagekit.service.in
@@ -4,7 +4,7 @@ Description=PackageKit Daemon
 # currently the design is to have dedicated daemons like
 # eos-updater and rpm-ostree, and gnome-software talks to those.
 ConditionPathExists=!/run/ostree-booted
-Wants=network-online.target
+After=network-online.target
 
 [Service]
 Type=dbus
-- 
2.42.0

